### PR TITLE
fix transfer/show token precise problem

### PR DESCRIPTION
**Pull request by afeiluo**
before:
image
image

after:

image
image

token contract address:
https://kovan.etherscan.io/token/0x19663c148a6dc8bd0864710603c29c2031f4a77a

### CODE REVIEW

**Review by yingjingyang — APPROVED** [review]

File: basic/09-hardhat-react/frontend/src/components/Dapp.js [hunk 1]
```diff
@@ -159,7 +160,8 @@ export class Dapp extends React.Component {
             {this.state.balance.gt(0) && (
               <Transfer
                 transferTokens={(to, amount) =>
-                  this._transferTokens(to, amount)
+                  // convert to contract precise amount
+                  this._transferTokens(to, amount*10**1)
```
Comment (sansx) [inline basic/09-hardhat-react/frontend/src/components/Dapp.js:164]: Do we need this `**1`? 🤔

**Comment by sansx** [comment]
the decimal about the transfer token should get dynamically which according to the token setting, https://docs.ethers.io/v5/api/contract/example/#erc20-meta-methods

**Review by 0x-stan — APPROVED** [review]

File: basic/09-hardhat-react/frontend/src/components/Dapp.js [hunk 2]
```diff
@@ -107,7 +107,8 @@ export class Dapp extends React.Component {
             <p>
               Welcome <b>{this.state.selectedAddress}</b>, you have{' '}
               <b>
-                {this.state.balance.toString()} {this.state.tokenData.symbol}
+                {/* show human read balance (deployed contract with precise 1 in /scripts/deploy.js) */}
+                {this.state.balance/10} {this.state.tokenData.symbol}
```
Comment (0x-stan) [inline basic/09-hardhat-react/frontend/src/components/Dapp.js:111]: should get decimal dynamically

**Comment by afeiluo** [comment]
> the decimal about the transfer token should get dynamically which according to the token setting, https://docs.ethers.io/v5/api/contract/example/#erc20-meta-methods

thx for you advice, I have opt this issue with this pr https://github.com/Dapp-Learning-DAO/Dapp-Learning/pull/821

**Pull request merged**